docs/spec/index.md

```diff
@@ -1,54 +1,105 @@
 
-## Field Definitions
+There are two types of files used in the TODS standard:
 
-### deadheads.txt
+- **Supplement files**, used to add, modify, and delete information from public GTFS files to model the operational service for internal purposes (with a `_supplement` filename suffix).
+- **TODS-Specific files**, used to model operational elements not currently defined in the GTFS standard.
 
-| **Field Name** | **Type** | **Required** | **Description** |
-| --- | --- | --- | --- |
-| deadhead_id | ID | Required | Identifies a deadhead. |
-| service_id | ID referencing [**calendar.service_id**](https://developers.google.com/transit/gtfs/reference#calendartxt) | Required | Identifies a set of dates when the deadhead is scheduled to take place. |
-| block_id | ID | Required | Identifies the block to which the deadhead belongs. |
-| shape_id | ID referencing [**shapes.shape_id**](https://developers.google.com/transit/gtfs/reference#shapestxt) | Optional | Identifies a geospatial shape that describes the vehicle travel path for a deadhead. |
-| to_trip_id | ID referencing [**trips.trip_id**](https://developers.google.com/transit/gtfs/reference#tripstxt) | Conditionally Required | Identifies the trip scheduled immediately following to the deadhead within the block_id. |
-| from_trip_id | ID referencing [**trips.trip_id**](https://developers.google.com/transit/gtfs/reference#tripstxt) | Conditionally Required | Identifies the trip scheduled immediately prior to the deadhead within the block_id. |
-| to_deadhead_id | ID referencing **deadheads.deadhead_id** | Conditionally Required | Identifies the deadhead scheduled immediately following the deadhead within the block_id. |
-| from_deadhead_id | ID referencing **deadheads.deadhead_id** | Conditionally Required | Identifies the deadhead scheduled immediately prior to the deadhead within the block_id. |
+### Files
 
-### ops_locations.txt
+| **File Name** | **Type** | **Description** |
+| --- | --- | --- |
+| trips_supplement.txt | Supplement | Supplements and modifies GTFS [trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt) with deadheads and other non-public trip information. |
+| stops_supplement.txt | Supplement | Supplements and modifies GTFS [stops.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stopstxt) with internal stop locations, waypoints, and other non-public stop information.|
+| stop_times_supplement.txt | Supplement | Supplements and modifies GTFS [stop_times.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stop_timestxt) with non-public times at which trips stop at locations, `stop_times` entries for non-public trips, and related information. |
+| routes_supplement.txt | Supplement | Supplements and modifies GTFS [routes.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#routestxt) with internal route identifiers and other non-public route identification. |
```

Comment:
Swiftly has a loose preference to make the file suffix be _tods_supplement rather than _supplement, given that there are a couple of optional columns that start with TODS_ in those supplement files. 

We would rather have multiple supplement files per original file than a single supplement file that feels "sparse", i.e. many columns are always null for many rows, because there's a different set of columns targeted at different reasons for supplementing

Reply:
> Swiftly has a loose preference to make the file suffix be _tods_supplement rather than supplement, given that there are a couple of optional columns that start with TODS in those supplement files.

Makes sense! I'll confer with some of the other working group stakeholders and get back to you.

> We would rather have multiple supplement files per original file than a single supplement file that feels "sparse", i.e. many columns are always null for many rows, because there's a different set of columns targeted at different reasons for supplementing

If I'm understanding this comment correctly, the suggestion proposes allowing a TODS dataset to have two distinct instances of a supplement file (e.g. `trips_supplement_1.txt` and `trips_supplement_2.txt`), such that, for example, one could have the addition of full rows, and another could have some updates to values, without needing to leave the unmodified column values of the applicable update rows blank.

I'm not a huge fan of this from two perspectives:
1. Producers would implicitly be asked to split the content into sub-files (at considerable development effort/expense), which may or may not even be possible in a particular production system, if they even do this at all.
2. Consumers would be required to inspect a directory for multiple sub-files, rather than being able to more easily error-check or report based on the presence of absence of a particular file.

Reply:
I have no objection to adopting the _tods_supplement name scheme. 

@mads14 could you clarify what the multiple supplement files would look like and how that would help swiftly?

Reply:
Thanks for the responses!

To clarify - my suggestion at this point is to rename the files to include _tods_supplement at the end e.g. trips_tods_supplement.txt. My suggestion is not to split up the current proposed supplemental trips file into multiple files.

I imagine that in the future there was a non-TODS extension of GTFS that also requires supplemental trips. 

For example only, I’m not proposing this at the moment: say in the future GTFS was additionally extended so that a substitute “snow schedule” could be placed in supplemental GTFS files and could be swapped in for regular service as needed. This use case would potentially require a different set of additional columns and the TODS_ specific fields would be meaningless/always left blank. Over time we could end up with lots of columns that are only valid for one specific case. I guess the implementers of the next GTFS extension could have a choice and will propose what they will (whether to combine "tods" and the "special schedule" supplements or not) if and when the spec is further extended, but my preferences is that the supplemental files for this new “snow schedule”/“special schedule” are new supplemental files and use their own extension.

Reply:
Got it, seems like a good change to me @mads14 !